### PR TITLE
[chore] Move dependencies control to .env file

### PR DESCRIPTION
**Pull request by julianocosta89**
# Changes

A couple of dependencies had to be updated in multiple places, this PR moves all of them to `.env` file.

## Merge Requirements

For new features contributions, please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Comment by julianocosta89** [comment]
`build_images` is not using docker compose 😞

I'm open to suggestions, because I strongly believe we should have a centralised place to manage the versions.

As of today we have to update 3 files every time the java agent updates.
2 places every time there is an update to opensearch or tracetest.

File: .env [hunk 1]
```diff
@@ -10,13 +10,16 @@ COLLECTOR_CONTRIB_IMAGE=otel/opentelemetry-collector-contrib:0.108.0
...
+OPENSEARCH_IMAGE_VERSION=2.16.0
+OPENSEARCH_IMAGE=opensearchproject/opensearch:${OPENSEARCH_IMAGE_VERSION}
 POSTGRES_IMAGE=postgres:16.4
 PROMETHEUS_IMAGE=quay.io/prometheus/prometheus:v2.54.1
-VALKEY_IMAGE=valkey/valkey:8.0-alpine
-# must also update the version arg in ./test/tracetesting/Dockerfile
-TRACETEST_IMAGE=kubeshop/tracetest:v1.5.2
+TRACETEST_IMAGE_VERSION=v1.5.4
+TRACETEST_IMAGE=kubeshop/tracetest:${TRACETEST_IMAGE_VERSION}
+VALKEY_IMAGE=valkey/valkey:7.2-alpine
```
Comment (rogercoll) [inline .env:19]: Is there any reason behind this version change?

```suggestion
VALKEY_IMAGE=valkey/valkey:8.0-alpine
```

**Review by rogercoll — APPROVED** [review]
.env FTW! I like the idea of controlling as much as possible the versions from a single file

**Comment by rogercoll** [comment]
> build_images is not using docker compose 😞

Could this be changed to use `docker compose build`? :thinking: I agree that relying on a single dependencies file would ease the maintenance

File: .env (hunk 1 above)
Comment (julianocosta89) [reply .env:19]: @rogercoll I've reverted the update because it seems `8.0` is `RC`.
The latest stable is `7.2`.

**Comment by julianocosta89** [comment]
Unfortunately I'm not a Github actions expert.
I've googled a bit around and I'm still not sure.

We have this in the GH marketplace: https://github.com/marketplace?query=docker+compose
But only 5 of those are `verified` and none of those 5 do what we need here.

@puckpuck any ideas/suggestions?

**Comment by julianocosta89** [comment]
I'm gonna close this one as I can't fix the GH actions right now.

**Pull request closed without merging**